File: su/src/main/java/org/sqlbuilder/su/SuModule.java
package org.sqlbuilder.su;

import org.sqlbuilder.common.Module;
import org.sqlbuilder.common.NotFoundException;

import java.io.IOException;

public class SuModule extends Module
{
	public static final String MODULE_ID = "sumo";

	protected SuModule(final String conf, final Mode mode)
	{
		super(MODULE_ID, conf, mode);
	}

	@Override
	protected void run()
	{
		assert props != null;

		try
		{
			switch (mode)
			{
				case PLAIN:
					new SuProcessor(props).run();
					break;

				case RESOLVE:
					new SuResolvingProcessor(props).run();
					break;

				case UPDATE:
					new SuUpdatingProcessor(props).run();
					break;

				case EXPORT:
				default:
			}
		}
		catch (IOException | ClassNotFoundException e)
		{
			e.printStackTrace();
		}
	}

	public static void turnOffLogging()
	{
		final String pathKey = "java.util.logging.config.file";
		final String pathValue = "logging.properties";
		System.setProperty(pathKey, pathValue);

		final String classKey = "java.util.logging.config.class";
		final String classValue = System.getProperty(classKey);
		if (classValue != null && !classValue.isEmpty())
		{
			System.err.println(classKey + " = " + classValue);
		}
	}

	public static void main(String[] args) throws NotFoundException
	{
		turnOffLogging();

		int i = 0;
		Mode mode = Mode.PLAIN;
		if (args[i].startsWith("-"))
		{
			mode = Mode.read(args[i++]);
		}
		String conf = args[i];
		new SuModule(conf, mode).run();
	}
}

File: su/src/main/java/org/sqlbuilder/su/SuModule.kt
package org.sqlbuilder.su

import org.sigma.core.Logging
import org.sqlbuilder.common.Module
import org.sqlbuilder.common.Module.Mode.Companion.read
import org.sqlbuilder.common.NotFoundException
import java.io.IOException
import java.util.logging.LogManager

class SuModule(
    conf: String,
    mode: Mode
) : Module(MODULE_ID, conf, mode) {

    override fun run() {
        checkNotNull(props)

        try {
            when (mode) {
                Mode.PLAIN   -> SuProcessor(props).run()
                Mode.RESOLVE -> SuResolvingProcessor(props).run()
                Mode.UPDATE  -> SuUpdatingProcessor(props).run()
                Mode.EXPORT  -> {}
                else         -> {}
            }
        } catch (e: IOException) {
            e.printStackTrace()
        } catch (e: ClassNotFoundException) {
            e.printStackTrace()
        }
    }

    companion object {

        const val MODULE_ID: String = "sumo"

        private fun setLogging() {
            try {
                Logging::class.java.getClassLoader().getResourceAsStream("logging.properties").use {
                     LogManager.getLogManager().readConfiguration(it)
                }
            } catch (e: IOException) {
                e.printStackTrace()
            }
        }

        fun turnOffLogging() {
            setLogging()

            val classKey = "java.util.logging.config.class"
            val classValue = System.getProperty(classKey)
            if (classValue != null && !classValue.isEmpty()) {
                System.err.println("$classKey = $classValue")
            }
        }

        @Throws(NotFoundException::class)
        fun main(args: Array<String>) {
            turnOffLogging()

            var i = 0
            var mode = Mode.PLAIN
            if (args[i].startsWith("-")) {
                mode = read(args[i++])
            }
            val conf = args[i]
            SuModule(conf, mode).run()
        }
    }
}
